src/Sparrow.Server/Platform/RavenProcess.cs

```diff
@@ -0,0 +1,209 @@
+            {
+                ravenProcess.Start(CancellationToken.None);
+                ravenProcess.ReadTo(outputDel);
+            }
+        }
+
+        private void ReadTo(StreamWriteDelegate outputDel)
+        {
+            var bytes = new byte[4096];
+            using (var fs = new FileStream(StandardOutAndErr, FileAccess.Read))
+            {
+                var read = fs.Read(bytes, 0, 4096);
+                while (read != 0)
+                {
+                    outputDel?.Invoke(new Span<byte>(bytes, 0, read));
+                    read = fs.Read(bytes, 0, 4096);
+                }
+            }
+        }
+
+        public Task<string> ReadToEndAsync()
+        {
+            using (var fs = new FileStream(StandardOutAndErr, FileAccess.Read))
+            using (var sr = new StreamReader(fs))
+            {
+                return sr.ReadToEndAsync();
+            }
+        }
+
+        private void Kill()
+        {
+            if (_pid != IntPtr.Zero)
+            {
+                var rc = Pal.rvn_kill_process(_pid, out var errorCode);
+                if (rc != PalFlags.FailCodes.Success)
+                    PalHelper.ThrowLastError(rc, errorCode, $"Failed to kill proc id={_pid.ToInt64()}. Command: '{StartInfo.FileName} {StartInfo.Arguments}'");
+            }
+        }
+
+        private void ReadLines(StreamReader sr, Action<object, string> lineOutputHandler, CancellationToken ctk)
+        {
+            while (ctk.IsCancellationRequested == false)
+            {
+                var lineTask = sr.ReadLineAsync();
+                if (lineTask.Wait(-1, ctk) == false)
+                    break;
+                var line = lineTask.Result;
+                OnLineOutput(line, ctk);
+                if (line == null)
+                    break;
+            }
+        }
+
+        public static void Execute(string command, string arguments, int pollingTimeoutInSeconds, EventHandler exitHandler, Action<object, string> lineOutputHandler, CancellationToken ctk)
+        {
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = command,
+                Arguments = arguments
+            };
+            using (var process = new RavenProcess { StartInfo = startInfo })
+            {
+                if (exitHandler != null)
+                    process.ProcessExited += exitHandler;
+                if (lineOutputHandler != null)
+                    process.LineOutput += lineOutputHandler;
+                process.Start(ctk);
+                using (var fs = new FileStream(process.StandardOutAndErr, FileAccess.Read))
+                using (var sr = new StreamReader(fs, Encoding.UTF8))
+                {
+                    while (ctk.IsCancellationRequested == false)
+                    {
+                        var rc = Pal.rvn_wait_for_close_process(process._pid, pollingTimeoutInSeconds, out var exitCode, out var errorCode);
```

Comment:
why is the first thing that you are doing is waiting for it to close?

Reply:
waiting to close with timeout.  polling...

Reply:
ok. will be fixed so instead of polling, this is general totoal waitForExitTimeoutInSeconds